### PR TITLE
Updated several node dependencies

### PR DESCRIPTION
**Pull request by austinhinderer**
Updated babel and loader, updates webpack to 3.x. Cleaned up webpack config file to match 2.0/3.0 webpack config structure. Added repo link to fix build warning.

### CODE REVIEW

**Comment by austinhinderer** [comment]
Fixes #23

**Comment by austinhinderer** [comment]
This needs a bit of work, I'm going to re-submit when ready.

File: .babelrc [hunk 1]
```diff
@@ -1,5 +1,10 @@
 {
-  "presets": ["es2015", "stage-0", "react"],
-  "plugins": ["add-module-exports", "transform-decorators-legacy", "transform-runtime"]
+  "presets": ["env", "stage-0", "react"],
```
Comment (austinhinderer) [inline .babelrc]: 'env' packages replaces 'es2015' due to deprecation of interface by Babel.

File: package.json [hunk 2]
```diff
@@ -2,33 +2,39 @@
   "name": "NeoLink",
   "version": "0.1.29",
   "description": "A Chrome Extension Wallet for the Neo Smart Economy",
+  "repository": {
```
Comment (austinhinderer) [inline package.json]: Fixes warning during builds.

File: .babelrc [hunk 3]
```diff
@@ -1,5 +1,10 @@
...
+  "plugins": ["add-module-exports", "transform-decorators-legacy", "transform-runtime"],
+  "env": {
+    "development": {
+      "presets": ["react-hmre"]
```
Comment (austinhinderer) [inline .babelrc]: Fixes bug when rendering in production context.

File: package.json [hunk 4]
```diff
@@ -2,33 +2,39 @@
...
+    "type": "git",
+    "url": "https://github.com/CityOfZion/NeoLink.git"
+  },
   "scripts": {
-    "build": "node utils/build.js",
+    "build": "NODE_ENV=production node utils/build.js",
```
Comment (austinhinderer) [inline package.json]: Force execution in production context for best asset performance client side.

File: package.json [hunk 5]
```diff
@@ -2,33 +2,40 @@
...
-    "html-loader": "^0.4.5",
-    "html-webpack-plugin": "2.24.1",
+    "html-loader": "^0.5.1",
+    "html-webpack-plugin": "^2.24.1",
     "postcss-loader": "^2.0.9",
     "preact-material-components": "^1.3.2",
-    "style-loader": "^0.13.0",
-    "webpack": "2.2.1",
-    "webpack-dev-server": "^2.3.0",
+    "react": "^16.2.0",
```
Comment (slipo) [inline package.json]: Why do we need React in dev? Shouldn't we either exclusively use Preact or React?

**Comment by slipo** [comment]
Code looks good. Just that react question. Also please merge dev in. I'm guessing you can just remake the lock file.

After yarn build I'm seeing the following:
yarnbuild

Any ideas there? Do you see that too?

**Comment by slipo** [comment]
Not finding any errors btw in the console with the issue above.

File: package.json (hunk 5 above)
Comment (austinhinderer) [reply package.json]: It is required as a peer dependency by some modules we use. If it's not there, we get 5 warnings in the build console. I figure it's better to suppress the warnings to keep any noise out of the build than to just ignore them.

**Comment by austinhinderer** [comment]
I see the above state after build as well. My guess is there is a problem with the Redux store communication in build. Since the current output before my change is a white page with an error message, I figured this would be an improvement and in the next round I can look at getting the functionality working from build.

**Pull request closed without merging**